### PR TITLE
[RecoveryServices.Backup]Help text updates to AzRecoveryServicesBackup cmdlets

### PR DESCRIPTION
**Pull request by hiaga**
## Description

Added help text updates to provide more insights and clarity on the cmdlets. Modified values to reflect the current cmdlet behaviour.

## Checklist

- [ Yes] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ Yes] The title of the PR is clear and informative
- [Yes ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [Yes ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [N/A ] If applicable, the changes made in the PR have proper test coverage
- [N/A ] For public API changes to cmdlets:
    - [ N/A] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [N/A ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/GetAzureRmRecoveryServicesBackupProtectionPolicy.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ public class GetAzureRmRecoveryServicesBackupProtectionPolicy : RSBackupVaultCmd
         /// Backup management type of the policy to be fetched
         /// </summary>
         [Parameter(ParameterSetName = WorkloadBackupMangementTypeParamSet, Position = 3,
-            Mandatory = true, HelpMessage = ParamHelpMsgs.Common.BackupManagementType)]
+            Mandatory = false, HelpMessage = ParamHelpMsgs.Common.BackupManagementType)]
```
Comment (VeryEarly) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/GetAzureRmRecoveryServicesBackupProtectionPolicy.cs:61]: WorkloadBackupMangementTypeParamSet and WorkloadParamSet are identical after this change, please remove WorkloadBackupMangementTypeParamSet  if it's not needed

File: src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md [hunk 2]
```diff
@@ -67,15 +67,15 @@ This command gets the protection policy named DefaultPolicy, and then stores it
...
 
 ```yaml
 Type: System.Nullable`1[Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models.BackupManagementType]
 Parameter Sets: WorkloadBackupManagementTypeParamSet
 Aliases:
-Accepted values: AzureVM, MARS, SCDPM, AzureBackupServer, AzureSQL, AzureStorage, AzureWorkload
+Accepted values: AzureVM, MARS, SCDPM, AzureBackupServer, AzureSQL, AzureStorage, AzureWorkload, MAB
 
-Required: True
+Required: true
```
Comment (sambitratha) [inline src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md:78]: why is this changed?

File: src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md [hunk 3]
```diff
@@ -135,7 +135,7 @@ Currently, only AzureVM, AzureFiles is supported.
 Type: System.Nullable`1[Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models.WorkloadType]
 Parameter Sets: WorkloadParamSet, WorkloadBackupManagementTypeParamSet
 Aliases:
-Accepted values: AzureVM, AzureSQLDatabase, AzureFiles, MSSQL
+Accepted values: AzureVM, AzureSQLDatabase, AzureFiles, MSSQL, FileFolder
```
Comment (sambitratha) [inline src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md:138]: FileFolder is not supported. please remove this

File: src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md (hunk 2 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md:78]: I have reverted this to the previous text, as this is no longer needed.

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/GetAzureRmRecoveryServicesBackupProtectionPolicy.cs (hunk 1 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/GetAzureRmRecoveryServicesBackupProtectionPolicy.cs]: reverted this changed due to same reason - we need both param sets.

File: src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md (hunk 2 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md]: resolved

File: src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md (hunk 3 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectionPolicy.md]: removed

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Container/GetAzureRmRecoveryServicesBackupContainer.cs [hunk 4]
```diff
@@ -41,7 +41,7 @@ public class GetAzureRmRecoveryServicesBackupContainer : RSBackupVaultCmdletBase
         [Parameter(Mandatory = false, Position = 2,
             HelpMessage = ParamHelpMsgs.Container.BackupManagementType)]
         [ValidateNotNullOrEmpty]
-        [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage")]
+        [ValidateSet("AzureVM", "MARS", "AzureWorkload", "AzureStorage")]
```
Comment (VeryEarly) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Container/GetAzureRmRecoveryServicesBackupContainer.cs:44]: sorry I didn't identify this one before but can you add "AzureSQL" back to the validation set?
it's breaking change to make validation set more exclusive.

It seems like you didn't introduce SDK changes which end up with calling different API version, why is this "AzureSQL" value not supported anymore?
